source/backend/api/Solvers/TakeInteractionSolver.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections.Generic;
+using Pims.Dal.Entities;
+
+namespace Pims.Api.Services
+{
+    public class TakeInteractionSolver : ITakeInteractionSolver
+    {
+        private readonly HashSet<string> LandActWithOwnership;
+
+        public TakeInteractionSolver()
+        {
+            LandActWithOwnership = new HashSet<string>() { "Crown Grant", "Transfer" };
+        }
+
+        /// <summary>
+        /// Evaluates if the takes passed as an argument will result in an property owned (Core Inventory).
+        /// Note: This method does not validate the take status, nor if the takes belong to the same property.
+        /// </summary>
+        /// <param name="takes"></param>
+        public bool ResultsInOwnedProperty(IEnumerable<PimsTake> takes)
```

Comment:
Wrote this one to maximize readability. Another function would perform similarly for the has rights